packages/runtime/src/runtime.ts

```diff
@@ -364,27 +333,22 @@ export class Runtime {
    * @param appId Application Id
    * @param flags Stateful smart contract transaction optional parameters (accounts, args..)
    * @param payFlags Transaction Parameters
-   * @param program TEAL code as string
    */
   optInToApp (accountAddr: string, appId: number,
     flags: SSCOptionalFlags, payFlags: TxParams): void {
     const appParams = this.getApp(appId);
-    if (appParams) {
-      this.createOptInTx(accountAddr, appId, payFlags, flags);
-      this.ctx.state = cloneDeep(this.store);
-      const account = this.assertAccountDefined(accountAddr, this.ctx.state.accounts.get(accountAddr));
-      account.optInToApp(appId, appParams);
-
-      // Execute approval program for Opt-In
-      this.run(this.getProgram(appId).approvalProgram, ExecutionMode.STATEFUL);
-      this.store = this.ctx.state;
-    } else {
-      throw new TealError(ERRORS.TEAL.APP_NOT_FOUND, { appId: appId, line: 'unknown' });
-    }
+    this.addCtxOptInTx(accountAddr, appId, payFlags, flags);
+    this.ctx.state = cloneDeep(this.store);
+    const account = this.assertAccountDefined(accountAddr, this.ctx.state.accounts.get(accountAddr));
+    account.optInToApp(appId, appParams);
+
+    // Execute approval program for Opt-In
```

Comment:
maybe we can remove this, [approval-program] is very clear

Reply:
ok